template/src/App.tsx

```diff
@@ -15,72 +13,28 @@ import { theme } from './styles';
 // NOTE: Change this boolean to true to render the Storybook view for development!
 const RENDER_STORYBOOK = false;
 
-interface State {
-  removeWhenYouveAddedStateManagement?: true;
-  /** The root store or stores to inject into the app. ONLY USED WITH MST / REDUX APPS */
-  // store: any; // if you use this, type it to your root store
-  /** The configured Apollo client. ONLY USED WITH APOLLO / GRAPHQL APPS */
-  // client: ApolloClient<InMemoryCache>;
-}
+const App = (): ReactElement | null => {
+  // hide the splashscreen on mount
+  useEffect(() => {
+    SplashScreen.hide();
+  }, []);
 
-class App extends Component<{}, State> {
-  public state = {};
-
-  private setupApp = (): void => {
-    try {
-      // Use this block to setup either the store or the apollo client
-      // const rootStore = await setupRootStore();
-      // this.setState({ store });
-      //
-      // const client = await setupApolloClient();
-      // this.setState({ client })
-    } catch (error) {
-      throw error;
-    } finally {
-      // Now that everything is hydrated, hide the splashscreen
-      SplashScreen.hide();
-    }
-  };
-
-  public componentDidMount = async (): Promise<void> => {
-    this.setupApp();
-  };
-
-  public renderNavigation = (): ReactElement => {
-    // Generate the root nav of the app based on presence of a user (restored in setup functions above)
+  const renderNavigation = (): ReactElement => {
```

Comment:
```suggestion
  const Navigation = (): ReactElement => {
```

Reply:
Personally I don't like to ever define something as a component unless it lives in its own file.
I like all of these "on the fly render conditionals" to separate them mentally.

Its not a big deal though, as its literally just a naming convention :)

Reply:
Typically I agree, components should live in separate files. Though, in a case like this where they are related, very small, and only used in a single place, it can sometimes be more overhead in understanding it when its separated.

Agree it's not a big deal since like you said components are just functions that return jsx. Totally your call on which way we go.